codec/reflectcodec/type_codec.go

```diff
@@ -204,6 +207,23 @@ func (c *genericCodec) size(value reflect.Value) (int, bool, error) {
 		}
 		return size, constSize, nil
 
+	case reflect.Map:
+		size := wrappers.IntLen
+		// add the length of each key and value
+		for _, key := range value.MapKeys() {
+			innerSize, _, err := c.size(key)
+			if err != nil {
+				return 0, false, err
+			}
+			size += innerSize
+			innerSize, _, err = c.size(value.MapIndex(key))
+			if err != nil {
+				return 0, false, err
+			}
+			size += innerSize
+		}
```

Comment:
I think this should be using the `constSize` optimization right? It seems to always be dropped right now.